app/controllers/course/experience_points_records_controller.rb

```diff
@@ -1,21 +1,38 @@
 # frozen_string_literal: true
 class Course::ExperiencePointsRecordsController < Course::ComponentController
-  load_resource :course_user, through: :course, id_param: :user_id
+  load_resource :course_user, through: :course, id_param: :user_id, except: [:read_all_exp, :download]
   load_and_authorize_resource :experience_points_record, through: :course_user,
-                                                         class: Course::ExperiencePointsRecord.name
+                                                         class: Course::ExperiencePointsRecord.name,
+                                                         except: [:read_all_exp, :download]
+
+  def read_all_exp
+    authorize!(:read_all_exp, @course)
```

Comment:
Why are you authorising to read the course?

Reply:
Nevermind i will refactor

Reply:
experiences now belong to the course, and hence the ability whether user can read that page (containing all the experience points) lies in the course-level..